src/otx/v2/adapters/torch/mmengine/mmdet/engine.py

```diff
@@ -0,0 +1,214 @@
+        inferencer = DetInferencer(
+            model=model,
+            weights=checkpoint,
+            device=device,
+        )
+
+        return inferencer(img, batch_size, **kwargs)
+
+    def export(
+        self,
+        model: torch.nn.Module | (str | Config) | None = None,
+        checkpoint: str | Path | None = None,
+        precision: str | None = "float32",  # ["float16", "fp16", "float32", "fp32"]
+        task: str | None = "ObjectDetection",
+        codebase: str | None = "mmdet",
+        export_type: str = "OPENVINO",  # "ONNX" or "OPENVINO"
+        deploy_config: str | None = None,  # File path only?
+        device: str = "cpu",
+        input_shape: tuple[int, int] | None = None,
+        **kwargs,
+    ) -> dict:
+        """Export a PyTorch model to a specified format for deployment.
+
+        Args:
+            model (Optional[Union[torch.nn.Module, str, Config]]): The PyTorch model to export.
+            checkpoint (Optional[Union[str, Path]]): The path to the checkpoint file to use for exporting.
+            precision (Optional[str]): The precision to use for exporting.
+                Can be one of ["float16", "fp16", "float32", "fp32"].
+            task (Optional[str]): The task for which the model is being exported. Defaults to "Classification".
+            codebase (Optional[str]): The codebase for the model being exported. Defaults to "mmdet".
+            export_type (str): The type of export to perform. Can be one of "ONNX" or "OPENVINO". Defaults to "OPENVINO"
+            deploy_config (Optional[str]): The path to the deployment configuration file to use for exporting.
+                File path only.
+            device (str): The device to use for exporting. Defaults to "cpu".
+            input_shape (Optional[Tuple[int, int]]): The input shape of the model being exported.
+            **kwargs: Additional keyword arguments to pass to the export function.
+
+        Returns:
+            dict: A dictionary containing information about the exported model.
+        """
+        return super().export(
+            model=model,
+            checkpoint=checkpoint,
+            precision=precision,
+            task=task,
+            codebase=codebase,
+            export_type=export_type,
+            deploy_config=deploy_config,
+            device=device,
+            input_shape=input_shape,
+            **kwargs,
+        )
+
+    def _update_codebase_config(self, codebase: str | None, task: str | None, deploy_config_dict: dict) -> None:
+        """Update specific codebase config.
+
+        Args:
+            codebase(str): mmX codebase framework
+            task(str): mmdeploy task
+            deploy_config_dict(dict): Config dict for deployment
+        """
+        codebase = codebase if codebase is not None else self.registry.name
+        codebase_config = {
+            "type": codebase,
+            "task": task,
+            "post_processing": {
```

Comment:
I agree that post processing params should be added further. But, I am not sure why this is codebase config... What does codebase mean?

Reply:
codebase mean model's codebase such as mmdet, mmseg, mmaction, etc. So this codebase_config affects some modules in mmdet during export procedure.

Reply:
I don't like `codebase` either. If possible, let's use human-friendly term in OTX code and translate to mmX language just at the feeding time, hopefully. :)
Please ignore this if it's not the case.

Reply:
I am sorry that I was failed be aware of codebase. Could you provide an example for this?

Reply:
Codebase_config means some hyper-parameters for exporting model from specific codebase such as mmdet, mmseg.
Most of detection model requires additional postprocessing(nms) but other model from other codebase do not require such additional hyper parameter. Therefore we should set hyperparameters related with postprocessing of detection model to codebase_config

Reply:
It is defined in mmdeploy, so we can't change name. MMDeploy cfg requires ir_config, codebase_config, backend_config